index.qmd

```diff
@@ -81,6 +86,20 @@ Finally, we discussion of Julia's potential to address current challenges in inf
+
+The main advantage of this approach is its performance. It enables the specialised solver to manage demanding calculations effectively, such as factoring large matrices and conducting iterative searches, using highly optimised, compiled code. This capability makes it feasible to solve complex real-world problems that would otherwise be too difficult to compute.
+
+A common practice in scientific computing is to access these solvers from high-level environments such as R, Julia, or Python [@bezanson_2017]. Through low-level C interfaces, the solvers can be "wrapped," allowing users to define models within an interactive, high-level language. However, the characteristics of this wrapper and the surrounding ecosystem are important factors that influence the accessibility, efficiency, and reproducibility of the research, as implementation details can vary significantly across different environments. 
+
+The R ecosystem, for example, requires a more manual approach. The interface is handled by the ipoptr package, which, having not been officially updated since 2012, is no longer distributed on the main CRAN repository. Installation, therefore, requires users to manually compile the wrapper from the Ipopt source code, a process that can be sensitive to the specific toolchain and libraries on the user's system. Beyond the installation challenges, the problem definition itself requires the user to implement separate functions for the objective, constraints, and their derivatives, as well as the sparse structure of the Jacobian and Hessian matrices. This manual specification process can lead to lengthy code and increase the potential for implementation errors.
+
+In contrast, the Julia programming language provides a different workflow for mathematical optimisation. The ecosystem is designed around composable packages: the modelling language JuMP.jl is used to state the problem, while a separate solver package, such as Ipopt.jl, provides the interface to the solver itself. Julia's built-in package manager streamlines the installation process by automatically downloading the necessary pre-compiled solver binaries. The JuMP syntax allows users to define NLPs using a symbolic representation that closely resembles the standard mathematical formulation. The framework can then automate the computation of sparse Jacobian and Hessian matrices required by the solver through automatic differentiation (AD). This level of abstraction from the solver's API streamlines the modelling process and permits the user to focus on the high-level mathematical formulation of the problem rather than on low-level implementation details.
```

Comment:
I think I was referring to all of this qas we will need to introduce these concepts earlier and so this would be duplicative. I think we can instead merge this as is and then edit as we add more further up